spec/classes/puppet_spec.rb

```diff
@@ -139,6 +139,25 @@
         end
       end#no params
 
+      context 'when allinone is true' do
+        let(:params) {{'allinone' => true}}
+        it 'should manage the facts directories' do
```

Comment:
Yep that's true I was slightly confused.
The no params test should stay the same because it doesn't include setting allinone and changing the directory ownership.
This test should be about the directory ownership and should be renamed to reflect that and have an alternate case when allinone is not set.